File: framework/languages/dotnet/src/Zlink.Framework/Contracts/Streams/IZLinkStream.cs
namespace Zlink.Framework.Contracts.Streams;

public interface IZLinkStream
{
    string SessionId { get; }

    RoutingId? RoutingId { get; }

    string? LocalAddr { get; }

    string? RemoteAddr { get; }

    /// <summary>
    ///     Writes a stream payload through the framework message boundary.
    /// </summary>
    bool Write(
        ZLinkMessage payload,
        SendFlags flags = SendFlags.None);

    ValueTask CloseAsync();
}

File: framework/languages/node/packages/framework/src/contracts/Streams/IZLinkStream.ts
import type { RoutingId, ZLinkMessage } from '../Common';

export interface ZLinkStream {
  readonly sessionId: string;
  readonly routingId?: RoutingId;
  readonly localAddr?: string;
  readonly remoteAddr?: string;
  write(payload: ZLinkMessage, flags?: number): boolean;
  close(signal?: AbortSignal): Promise<void>;
}
